src/Umbraco.Web.UI.Client/src/common/resources/datatype.resource.js

```diff
@@ -30,369 +30,412 @@ function dataTypeResource($q, $http, umbDataFormatter, umbRequestHelper) {
+            "PostCreateCustomListView",
+            { contentTypeAlias: contentTypeAlias }
+          )),
+        "Failed to create a custom listview datatype");
+    },
+
+    /**
+     * @ngdoc method
+     * @name umbraco.resources.dataTypeResource#save
+     * @methodOf umbraco.resources.dataTypeResource
+     *
+     * @description
+     * Saves or update a data type
+     *
+     * @param {Object} dataType data type object to create/update
+     * @param {Array} preValues collection of prevalues on the datatype
+     * @param {Bool} isNew set to true if type should be create instead of updated
+     * @returns {Promise} resourcePromise object.
+     *
+     */
+    save: function (dataType, preValues, isNew) {
+
+      var saveModel = umbDataFormatter.formatDataTypePostData(dataType, preValues, "save" + (isNew ? "New" : ""));
+
+      return umbRequestHelper.resourcePromise(
+        $http.post(umbRequestHelper.getApiUrl("dataTypeApiBaseUrl", "PostSave"), saveModel),
+        "Failed to save data for data type id " + dataType.id);
+    },
+
+    /**
+     * @ngdoc method
+     * @name umbraco.resources.dataTypeResource#move
+     * @methodOf umbraco.resources.dataTypeResource
+     *
+     * @description
+     * Moves a node underneath a new parentId
+     *
+     * ##usage
+     * <pre>
+     * dataTypeResource.move({ parentId: 1244, id: 123 })
+     *    .then(function() {
+     *        alert("node was moved");
+     *    }, function(err){
+     *      alert("node didnt move:" + err.data.Message); 
+     *    });
+     * </pre> 
+     * @param {Object} args arguments object
+     * @param {Int} args.idd the ID of the node to move
+     * @param {Int} args.parentId the ID of the parent node to move to
+     * @returns {Promise} resourcePromise object.
+     *
+     */
+    move: function (args) {
+      if (!args) {
+        throw "args cannot be null";
+      }
+      if (!args.parentId) {
+        throw "args.parentId cannot be null";
+      }
+      if (!args.id) {
+        throw "args.id cannot be null";
+      }
+
+      return umbRequestHelper.resourcePromise(
+        $http.post(umbRequestHelper.getApiUrl("dataTypeApiBaseUrl", "PostMove"),
+          {
+            parentId: args.parentId,
+            id: args.id
+          }, { responseType: 'text' }),
+        'Failed to move content');
+    },
+
+    /**
+     * @ngdoc method
+     * @name umbraco.resources.dataTypeResource#copy
+     * @methodOf umbraco.resources.dataTypeResource
+     *
+     * @description
+     * Copies a node underneath a new parentId
+     *
+     * ##usage
+     * <pre>
+     * dataTypeResource.copy({ parentId: 1244, id: 123 })
+     *    .then(function() {
+     *        alert("node has been copied");
+     *    }, function(err){
+     *      alert("node didnt copy:" + err.data.Message);
+     *    });
+     * </pre>
+     * @param {Object} args arguments object
+     * @param {Int} args.idd the ID of the node to copy
```

Comment:
Small typo probably: `args.id` instead of `args.idd` 😉

Reply:
Fixed!